pkg/rbd/rbd_util.go

```diff
@@ -107,64 +108,95 @@ var (
 
 // createImage creates a new ceph image with provision and volume options.
 func createImage(ctx context.Context, pOpts *rbdVolume, volSz int64, cr *util.Credentials) error {
-	var output []byte
-
-	image := pOpts.RbdImageName
 	volSzMiB := fmt.Sprintf("%dM", volSz)
+	options := librbd.NewRbdImageOptions()
 
 	logMsg := "rbd: create %s size %s format 2 (features: %s) using mon %s, pool %s "
 	if pOpts.DataPool != "" {
 		logMsg += fmt.Sprintf("data pool %s", pOpts.DataPool)
+		err := options.SetString(librbd.RbdImageOptionDataPool, pOpts.DataPool)
+		if err != nil {
+			return err
+		}
 	}
 	klog.V(4).Infof(util.Log(ctx, logMsg),
-		image, volSzMiB, pOpts.ImageFeatures, pOpts.Monitors, pOpts.Pool)
+		pOpts.RbdImageName, volSzMiB, pOpts.ImageFeatures, pOpts.Monitors, pOpts.Pool)
 
-	args := []string{"create", image, "--size", volSzMiB, "--pool", pOpts.Pool, "--id", cr.ID, "-m", pOpts.Monitors, "--keyfile=" + cr.KeyFile, "--image-feature", pOpts.ImageFeatures}
+	if pOpts.ImageFeatures != "" {
+		// TODO: parse imagefeatures and set RbdImageOptionFeatures
+		// err = options.SetUint64(librbd.RbdImageOptionFeatures, 0)
+		klog.Infof(util.Log(ctx, "ImageFeatures (set to \"%v\" are currently "+
+			"ignored"), pOpts.ImageFeatures)
+	}
 
-	if pOpts.DataPool != "" {
-		args = append(args, "--data-pool", pOpts.DataPool)
+	ioctx, err := pOpts.getIoctx(cr)
+	if err != nil {
+		return err
 	}
-	output, err := execCommand("rbd", args)
+	defer ioctx.Destroy()
 
+	image, err := librbd.CreateWithOptions(ioctx, pOpts.RbdImageName, uint64(volSz), options)
 	if err != nil {
-		return errors.Wrapf(err, "failed to create rbd image, command output: %s", string(output))
+		return errors.Wrapf(err, "failed to create rbd image")
 	}
 
+	// image is not returned to the caller
+	// no need to Close(), the image is not open after Create()
+	_ = image
```

Comment:
just curios, what is the meaning of this line?

Reply:
This prevents Go from complaining that the `image` variable is unused.

Reply:
Why is an "image" being returned?

Reply:
It is not needed, I'll drop it and move the comments to just before the `librbd.Create4()` call.